wfsim/core/pulse.py

```diff
@@ -196,6 +210,41 @@ def uniform_to_pe_arr(self, p, channel=0):
         indices = (p * 2000).astype(np.int64) + 1
         return self.__uniform_to_pe_arr[channel, indices]
 
+    def add_truth(self,
+                  photon_timings,
+                  photon_gains,
+                  dt,
+                  channel,
+                  n_double_pe,
+                 ):
+
+        if channel in self.config['turned_off_pmts']:
+            return
+
+        if str(channel) in self.config.get('special_thresholds', {}):
```

Comment:
Sorry can you remind me what is `special_thresholds`? ~~And where is the rounding that makes us want to do the `-0.5`?~~ No worries I figured out. The rounding might be from digitization

Reply:
Special channels are those not set to 15 ADC, usually more to reduce noise triggers.